portafly/doc/ux_specs/global_components/alerts.md

```diff
@@ -0,0 +1,24 @@
+# Alerts UX/UI specs
+
+[Back to index](../index.md)
+
+* [PatternFly Alerts design guidelines](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/alerts-and-notifications)
+* [PatternFly-react Alerts docs](https://www.patternfly.org/v4/documentation/react/components/alert)
+
+### Network error alerts
+* Network errors are displayed in **toast alerts** ([PF specs](https://www.patternfly.org/v4/documentation/react/components/alert#types)), unless the current state of the UI is blocking users' view of the page layout (eg. modal dialogs)
+  * In this case, the errors will be displayed as **inline alerts** ([PF specs](https://www.patternfly.org/v4/documentation/react/components/alert#inline-types)), making sure the positioning is as close as possible to the trigger (eg. button) of the action that launched the network request.
```

Comment:
@josemigallas is this even a possible scenario? What is it exactly that is dismissing the modal, clicking on the button that submits the form or that the request returns successfully? In the latter scenario, what if the request takes some time, will the user be stuck in the modal?

Reply:
It's a perfectly valid scenario IMO. The model would be dismissed after the action is successful (server responded with an OK).

The user should be stuck in the modal if there's an error to avoid losing information (forms) and having to open the same modal again (which I as an user find annoying).

Reply:
@josemigallas it's tricker than that. locking the user into the modal is bad UX. for instance, what if for some reason the server takes minutes to respond? the user will be stuck in the modal view, whereas they could instead submit a bulk action on a large number of objects and then navigate away to their next task while the bulk action is processed in the background.

at the same time, yes, I agree that we don't want them to eventually lose the content they entered in the modal before we're sure the server responded successfully.

what do you think about the following proposals?
- The modal is dismissed as soon as the request to the server is **sent**. 
- A toast alert will be displayed showing the status of the request.
- User-generated content is temporarily saved in the fields and will be deleted **only** when the response from the server is OK.
- In case an error is triggered, the toast alert will update to display a button that says something like "Try again". Clicking on the button will reopen the modal in the last state before it was dismissed (pre-selected items, user-generated content) – If user dismisses the toast alert the user-generated content will be deleted as well.